docker-compose.yaml

```diff
@@ -1,128 +1,90 @@
-      # - "--reverse-proxy=true"
-      # - "--redirect-url=https://auth.fsn1.<REDACTED>/oauth2/callback"
-      # - "--scope=user:email"
-      # - "--set-xauthrequest=true"
-      - "--upstream=static://202"
-      # - "--whitelist-domain=.<REDACTED>"
-      # - "--whitelist-domain=.fsn1.<REDACTED>"
-      - "--skip-jwt-bearer-tokens=true"
     environment:
+      # service settings
       OAUTH2_PROXY_HTTP_ADDRESS: 0.0.0.0:4180
-      OAUTH2_PROXY_PROVIDER: oidc
-      OAUTH2_PROXY_PROVIDER_DISPLAY_NAME: Keeycloak
-      OAUTH2_PROXY_OIDC_ISSUER_URL: https://keycloak.${IP:-128.208.230.197}.nip.io/auth/realms/fEMR
-      OAUTH2_PROXY_EMAIL_DOMAINS: "*"
-      OAUTH2_PROXY_INSECURE_OIDC_ALLOW_UNVERIFIED_EMAIL: "true"
-      OAUTH2_PROXY_USER_ID_CLAIM: "preferred_username"
-      OAUTH2_PROXY_SCOPE: openid profile email
-      OAUTH2_PROXY_COOKIE_DOMAINS: .${IP:-128.208.230.197}.nip.io
-      OAUTH2_PROXY_WHITELIST_DOMAINS: .${IP:-128.208.230.197}.nip.io
+      OAUTH2_PROXY_REVERSE_PROXY: "true"
+      OAUTH2_PROXY_SET_XAUTHREQUEST: "true"
+      OAUTH2_PROXY_SET_AUTHORIZATION_HEADER: "true"
       OAUTH2_PROXY_PASS_AUTHORIZATION_HEADER: "true"
-      OAUTH2_PROXY_PASS_ACCESS_TOKEN: "true"
       OAUTH2_PROXY_PASS_USER_HEADERS: "true"
-      OAUTH2_PROXY_SET_AUTHORIZATION_HEADER: "true"
-      OAUTH2_PROXY_SET_XAUTHREQUEST: "true"
-      OAUTH2_PROXY_COOKIE_REFRESH: 1m
+      OAUTH2_PROXY_PASS_ACCESS_TOKEN: "true"
+      # allow validation of JWTs in Authorization HTTP header
+      # https://medium.com/in-the-weeds/service-to-service-authentication-on-kubernetes-94dcb8216cdc#20d0
+      OAUTH2_PROXY_SKIP_JWT_BEARER_TOKENS: "true"
+
+      # when authenticated, return a static 202 response
+      # https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview/#forwardauth-with-static-upstreams-configuration
+      OAUTH2_PROXY_UPSTREAMS: static://202
+
+      # general cookie settings
+      OAUTH2_PROXY_EMAIL_DOMAINS: "*"
       OAUTH2_PROXY_COOKIE_EXPIRE: 30m
-      OAUTH2_PROXY_COOKIE_SECRET: OQINaROshtE9TcZkNAm-5Zs2Pv3xaWytBmc5W7sPX7w=
-      # OAUTH2_PROXY_REDIRECT_URL: http://auth-proxy-${COMPOSE_PROJECT_NAME}:4180/oauth2/callback
-      # OAUTH2_PROXY_REDIRECT_URL: http://oauth:4180/oauth2/callback
-      # OAUTH2_PROXY_REDIRECT_URL: https://${SERVER_NAME}/oauth2/callback
-      OAUTH2_PROXY_REDIRECT_URL: https://oauth2-proxy.${IP:-128.208.230.197}.nip.io/oauth2/callback
-      OAUTH2_PROXY_CLIENT_ID: auth_proxy_test
-      OAUTH2_PROXY_CLIENT_SECRET: 74c6fdc4-4086-4a21-bf40-49c7ee74357e
-      OAUTH2_PROXY_REVERSE_PROXY: "true"
-    expose:
-      - '4180'
+      OAUTH2_PROXY_COOKIE_REFRESH: 1m
+      OAUTH2_PROXY_INSECURE_OIDC_ALLOW_UNVERIFIED_EMAIL: "true"
+      # claim to base session cookie on; Keycloak account email not always set
+      OAUTH2_PROXY_USER_ID_CLAIM: preferred_username
+
+      # OIDC integration settings
+      OAUTH2_PROXY_PROVIDER: oidc
+      OAUTH2_PROXY_PROVIDER_DISPLAY_NAME: Keeycloak
```

Comment:
did you intend to leave that extra 'e' in?

Reply:
Thanks for pointing that out! I left it as an example, but env vars cannot be overridden when set in the `environment` section of dc YAML. 

I've removed that env var from the `environment` section in this PR, but made sure to add a "stub" value in the corresponding `env_file` (`auth-proxy.env`) in the [helloworld-environments PR](https://github.com/uwcirg/helloworld-environments/pull/5)